src/DataTable/__tests__/DataTable.test.tsx

```diff
@@ -262,4 +272,259 @@ describe('DataTable', () => {
       expect(screen.getByRole('rowheader', {name: row.name})).toBeInTheDocument()
     }
   })
+
+  describe('sorting', () => {
```

Comment:
I went ahead and removed the custom sorting behavior so that they could be added through: https://github.com/github/primer/issues/1833 since I think you're dead-on that they should have examples plus tests and I did not include that in this PR